### PR TITLE
JAMES-2305 Update to javax.mail 1.6.0

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Comment by chibenwa** [comment]
```
[5d7ff572212788fe00aea8ef958685f928c3e05d] [INFO] Apache James :: Standard Mailets ................... FAILURE [ 37.674 s]
```

... major API changes they said...

**Comment by chibenwa** [comment]
No, that's why I ended up doing regex....

If you find better, tell me...

**Comment by chibenwa** [comment]
test this please

File: mailet/standard/src/test/java/org/apache/james/transport/mailets/AddFooterTest.java [hunk 1]
```diff
@@ -132,28 +132,21 @@ public void shouldEnsureCarriageReturnWhenAddFooterWithTextPlainMessage() throws
...
-                "Test=E0 and one =A4",
-                "------ my footer =E0/=A4 ------");
+        String expectedFooter = "------ " + MY_FOOTER + " =E0/=A4 ------";
         
 
         Mail mail = FakeMail.fromMime(quotedPrintableTextPlainMessage, javaCharset, javaMailCharset);
         mailet.service(mail);
 
-        assertThat(MailUtil.toString(mail, javaCharset)).isEqualTo(expected);
+        assertThat(MailUtil.toString(mail, javaCharset)).endsWith(expectedFooter);
```
Comment (mbaechler) [inline mailet/standard/src/test/java/org/apache/james/transport/mailets/AddFooterTest.java]: the test is about carriage return, what you don't check anymore

Comment (chibenwa) [reply mailet/standard/src/test/java/org/apache/james/transport/mailets/AddFooterTest.java]: Sorry. It is fixed...

**Comment by chibenwa** [comment]
test this please

**Comment by rouazana** [comment]
test this please
DefaultMailboxesProvisioningFilterTest.createMailboxesIfNeededShouldNotGenerateExceptionsInConcurrentEnvironment:83

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**